mephisto/abstractions/architects/ec2/ec2_architect.py

```diff
@@ -156,6 +158,17 @@ def download_file(self, target_filename: str, save_dir: str) -> None:
                 if chunk:
                     out_file.write(chunk)
 
+    def check_domain_exists(self, subdomain_name: str):
+        """
+          Checks if the domain name exists in Route53
+        """
+        with open(self.server_detail_path, "r") as server_detail_file:
+            server_detail_file_files = server_detail_file.read().split('/n')
+            for file in server_detail_file_files:
+                if file.startswith(f'{subdomain_name}.json'):
+                    return True
+        return False
```

Comment:
The check itself is really much simpler - we just need to see if the file at `self.server_detail_path` exists or not. If not, we're clear to create. If so, then there's potentially an issue.